master/internal/api_workspace.go

```diff
@@ -1055,31 +1054,17 @@ func (a *apiServer) DeleteWorkspace(
 		if err != nil {
 			return nil, fmt.Errorf("error deleting workspace (%d): %w", req.Id, err)
 		}
-
-		// Delete the auto-generated namespace (if it exists) and its resources in Kubernetes.
-		err = a.m.rm.DeleteNamespace(*namespaceName)
-		if err != nil {
-			return nil, err
-		}
-		for _, v := range deletedBindings {
-			err = a.m.rm.RemoveEmptyNamespace(v.Namespace, v.ClusterName)
-			if err != nil {
-				return nil, err
-			}
-		}
-
-		return &apiv1.DeleteWorkspaceResponse{Completed: true}, nil
+	} else {
+		go func() {
+			a.deleteWorkspace(ctx, req.Id, projects)
+		}()
 	}
```

Comment:
Can we change this section to 
```
go func() {
        a.deleteWorkspace(ctx, req.Id, projects)
}()

... DeleteNamespace
... for _, v := range toBeDeletedBindings{...}

if len(projects) == 0 {
       return &apiv1.DeleteWorkspaceResponse{Completed: true}, nil
}
 return &apiv1.DeleteWorkspaceResponse{Completed: false}, nil
```
Since we call `err = a.m.db.QueryProto("delete_workspace", holder, req.Id)` in `deleteWorkspace`, we really only need to know if the project length is 0 to determine if we want to set `Completed` to `true`!

Reply:
done